torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -2051,8 +2051,17 @@ int64_t AtenXlaType::size(const at::Tensor& self, int64_t dim) const {
 
 at::Tensor AtenXlaType::slice(const at::Tensor& self, int64_t dim,
                               int64_t start, int64_t end, int64_t step) const {
+  if (step != 1) {
+    return AtenXlaTypeBase::slice(self, dim, start, end, step);
+  }
+  XLATensor self_tensor = bridge::GetXlaTensor(self);
```

Comment:
I'd keep the `XLATensor::slice` and do the `dim`, `start` and `end` canonicalization there and keep this trivial.

Reply:
SG